datasetinsights/datasets/synthetic.py

```diff
@@ -72,3 +85,182 @@ def read_bounding_box_2d(annotation, label_mappings=None):
+
+        Returns (Tuple(Image,List(BBox2D))): Tuple comprising the image and
+        bounding boxes found in that image with transforms applied.
+
+        """
+        cap = self.catalog.iloc[index]
+        capture_file = cap.filename
+        ann = cap["annotation.values"]
+
+        capture = Image.open(os.path.join(self._data_path, capture_file))
+        capture = capture.convert("RGB")  # Remove alpha channel
+        annotation = read_bounding_box_2d(ann, self.label_mappings)
+
+        if self.transforms:
+            capture, annotation = self.transforms(capture, annotation)
+
+        return capture, annotation
+
+    def __len__(self):
+        return len(self.catalog)
+
+    def _cleanup(self, catalog):
+        """
+        remove rows with captures that having missing files and remove examples
+        which have no annotations i.e. an image without any objects
+        Args:
+            catalog (pandas dataframe):
+
+        Returns: dataframe without rows corresponding to captures that have
+        missing files and removes examples which have no annotations i.e. an
+        image without any objects.
+
+        """
+        catalog = self._remove_captures_with_missing_files(
+            self._data_path, catalog
+        )
+        catalog = self._remove_captures_without_bboxes(catalog)
+
+        return catalog
+
+    @staticmethod
+    def _remove_captures_without_bboxes(catalog):
+        """Remove captures without bounding boxes from catalog
+
+        Args:
+            catalog (pd.Dataframe): The loaded catalog of the dataset
+
+        Returns:
+            A pandas dataframe with empty bounding boxes removed
+        """
+        keep_mask = catalog["annotation.values"].apply(len) > 0
+
+        return catalog[keep_mask]
+
+    @staticmethod
+    def _remove_captures_with_missing_files(root, catalog):
+        """Remove captures where image files are missing
+
+        During the synthetic dataset download process, some of the files might
+        be missing due to temporary http request issues or url corruption.
+        We should remove these captures from catalog so that it does not
+        stop the training pipeline.
+
+        Args:
+            catalog (pd.Dataframe): The loaded catalog of the dataset
+
+        Returns:
+            A pandas dataframe of the catalog with missing files removed
+        """
+
+        def exists(capture_file):
+            path = Path(root) / capture_file
+
+            return path.exists()
+
+        keep_mask = catalog.filename.apply(exists)
+
+        return catalog[keep_mask]
+
+    @staticmethod
+    def _preprocess_dataset(data_path):
+        """ Preprocess dataset inside data_path and un-archive if necessary.
+
+        Args:
+            data_path (str): Path where dataset is stored.
+
+        Return:
+            Path of the dataset files.
+        """
+        archive_file = Path(data_path) / SynDetection2D.ARCHIVE_FILE
+        if archive_file.exists():
```

Comment:
@86sanj Why do we need to unarchive files for the dataset generated by perception package?